File: leetcode/1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        winner_counter = Counter(match[0] for match in matches)
        loser_counter = Counter(match[1] for match in matches)

        gigachad = [kid for kid in winner_counter if kid not in loser_counter]
        get_better = [kid for kid, count in loser_counter.items() if count == 1]

        gigachad.sort()
        get_better.sort()
        return [gigachad, get_better]

File: leetcode/1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        unordered_map<int, int> winners, losers;
        for (const auto& x : matches) {
            winners[x[0]]++;  
            losers[x[1]]++;  
        }
        
        vector<int> neverLost, lostOnce;
        for (const auto& w : winners) {
            if (losers.find(w.first) == losers.end()) {
                neverLost.push_back(w.first);
            }
        }
        for (const auto& l : losers) {
            if (l.second == 1) { 
                lostOnce.push_back(l.first);
            }
        }

        ranges::sort(neverLost), ranges::sort(lostOnce);
        return {neverLost, lostOnce};
    }
};